sklearn/calibration.py

```diff
@@ -338,22 +338,35 @@ def fit(self, X, y, sample_weight=None):
             else:
                 this_estimator = clone(base_estimator)
                 _, method_name = _get_prediction_method(this_estimator)
-                pred_method = partial(
-                    cross_val_predict,
-                    estimator=this_estimator,
-                    X=X,
-                    y=y,
-                    cv=cv,
-                    method=method_name,
-                    n_jobs=self.n_jobs,
-                )
-                predictions = _compute_predictions(
-                    pred_method, method_name, X, n_classes
-                )
 
                 if sample_weight is not None and supports_sw:
+                    pred_method = partial(
```

Comment:
I assume that you could simplify the call with a smaller diff:

```python
                    fit_params = (
                        {"sample_weight": sample_weight}
                        if sample_weight is not None and supports_sw
                        else None
                    )
                    pred_method = partial(
                        cross_val_predict,
                        estimator=this_estimator,
                        X=X,
                        y=y,
                        cv=cv,
                        method=method_name,
                        n_jobs=self.n_jobs,
                        fit_params={"sample_weight": sample_weight},
                    )
```

Reply:
agreed

Reply:
agreed, but with ```fit_params=fit_params``` in the ```partial```.